core/services/workflows/engine.go

```diff
@@ -217,7 +216,20 @@ func (e *Engine) loop(ctx context.Context) {
 				continue
 			}
 
-			err := e.startExecution(ctx, resp.Value)
+			te := &capabilities.TriggerEvent{}
+			err := resp.Value.UnwrapTo(te)
+			if err != nil {
+				e.logger.Errorf("could not unwrap trigger event", resp.Err)
+				continue
+			}
+
+			executionID, err := generateExecutionID(e.workflow.id, te.ID)
+			if err != nil {
+				e.logger.Errorf("could not generate execution ID", resp.Err)
+				continue
+			}
+
+			err = e.startExecution(ctx, executionID, resp.Value)
```

Comment:
This is relevant to Devidas' comment on my PR in common. Can we pass "te.Payload" here instead of "resp.Value"?

Reply:
Hmmmm... why? resp.Value needs to be used for interpolation (it's added to the execution state, see https://github.com/smartcontractkit/chainlink/pull/12703/files/b63a252712b24f99c8d9f1ac6280dc4978db08aa#diff-b1ca9fad3006c72cc7e4401eb80d180a603350c0cbcdfc121b90ae88172742cdR282